src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirstNode(value);
+        } else {
+            Node newNode = new Node<>(tail, value, null);
+            newNode.prevNode.nextNode = newNode;
+            tail = newNode;
+            size++;
+        }
+
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            addFirstNode(value);
+            return;
+        }
+        checkIndex(index);
+        // add value in the middle
+        Node currentNode = getNode(index);
+        Node newNode = new Node<>(currentNode.prevNode, value, currentNode);
+        newNode.prevNode.nextNode = newNode;
+        currentNode.prevNode = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listItem : list) {
+            add(listItem);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
```

Comment:
delete this line
![image](https://github.com/mate-academy/jv-linked-list/assets/20326858/9607b00d-42a2-4ed8-bf0a-d4c3ca2d3233)